docs/databases/postgresql/how-to-deploy-ha-postgresql-cluster.md

```diff
@@ -0,0 +1,408 @@
+
+This guide will explain how you can set up a highly available Postgres cluster of three servers using [Patroni](https://github.com/zalando/patroni).
+
+## Before You Begin
+
+This guide uses Ubuntu 16.04 and assumes familiarity with its package manager and user management.
+
+{: .note}
+>
+> This guide is written for a non-root user. Commands that require elevated privileges are prefixed with `sudo`. If you’re not familiar with the `sudo` command, see the [Users and Groups](/docs/tools-reference/linux-users-and-groups) guide.
+
+## A Brief Introduction to the Tools Being Used
+
+In this guide, we will create a highly available Postgres cluster that will failover automatically in case the Postgres master goes down.
+
+This setup will involve the installation and configuration of:
+
+- Postgres (on three Linodes)
+- [etcd](https://coreos.com/etcd) (on one Linode)
+- Patroni (on each of the Postgres Linodes)
+- [HAProxy](https://www.haproxy.org/) (on one Linode)
+
+### etcd
+
+etcd is a fault-tolerant, distributed key-value store that is used to store the state of the Postgres cluster. All the Postgres nodes, via Patroni, make use of etcd to keep the Postgres cluster up and running.
+
+### Patroni
+
+Patroni is an open source python package that manages Postgres configuration. It is installed on the same servers as Postgres and can be configured to handle tasks like replication, backups and restorations.
+
+Patroni, as used in this guide, will:
+
+- Configure the Postgres instance running on the same server
+- Configure replication from master to slaves
+- Automatically failover to the best slave in case the master goes down.
+
+### HAProxy
+
+When developing an application that uses a database, it can be cumbersome to keep track of the database endpoints if they keep changing. Using HAProxy simplifies this by giving us a single endpoint that we can connect our application to.
+
+HAProxy takes care of forwarding the connection to whichever node is currently the master. It does this by using a REST endpoint that Patroni provides. Patroni ensures that, at any given time, only one Postgres node (the master) will appear as online, forcing HAProxy to connect to the correct node.
+
+## Required Tools
+
+This section will guide you through the steps needed to install the above mentioned tools.
+
+### Install PostgreSQL
+
+1. On three Linodes where you want Postgres installed, begin by updating the package lists:
+
+        sudo apt-get update
+
+2. Next, install Postgres:
+
+        sudo apt-get install postgresql-9.5 -y
+
+3. When you install Postgres, it also starts running as a service. We need to stop Postgres so that Patroni can manage it from this point on:
+
+        sudo systemctl stop postgresql
+
+4. Patroni manages utilities that comes default with Postgres. In Ubuntu 16.04, these utilities are installed in the `/usr/lib/postgresql/9.5/bin` directory by default.
```

Comment:
This should be something to the effect of "Patroni uses some utilities that comes with Postgres, to manage Postgres"